### DISCUSSION TITLE
Making a Responsive Form - Pulling Information from separated Table

### DISCUSSION THREAD

**Discussion by smtschampel** [Budibase/budibase#11800] [Help]
Hello everyone,

I'm seeking guidance and assistance in creating a responsive form that will be used to record and manage trip data for a trucking company. This form is crucial for efficiently documenting trip details, and I want to ensure that it provides accurate and up-to-date information about each route.

**Form Overview**

Form Fields: The form will include various fields to capture essential trip data, such as:

Date and time of departure and arrival
Driver information
Truck details
Cargo information
Route selection
Additional notes, if needed
Route Selection: One of the key features of this form is the "Route Selection" option. Users can choose the route the truck will take from a list of predefined routes. The route information, including distance, is stored in a separate database table.

Dynamic Distance Retrieval: To enhance the form's usability, I'd like to make it dynamically retrieve and display the distance associated with the selected route as soon as the user makes a selection.

**Current Challenge**

I've already made some progress using the following code snippet:

{{#each Ruta.Rows }}
 {{#if (eq  Field Value  this.Ruta)}}
      {{this.Distancia}}
  {{/if}}
{{/each}}

However, I've encountered an issue where the code doesn't work as expected when using the "Field Value" variable. It does work when I manually input the route text, like 'Barranquilla-SantaMarta'. I've also verified that the "Field Value" is correct before it enters this code block, but the comparison seems to break within the loop.

**Seeking Guidance**

I'd appreciate your expertise and advice on how to resolve this issue and improve the form's overall functionality. Additionally, if you have any general guidelines or best practices for creating such forms with dynamic data retrieval, please feel free to share them. Your input is invaluable in ensuring that this form operates smoothly and efficiently.

Thank you for your help!

**Comment by ConorWebb96** [comment]
Hey @smtschampel,

Are you using the Internal Budibase data source?

Heres some documentation that may help you in this case;

https://docs.budibase.com/docs/filter-by-relationships#tutorial-filter-on-selected-relationship-picker-values
https://docs.budibase.com/docs/fetching-data#displaying-relational-data

I hope this helps.

**Reply by smtschampel** [reply]
Hi @ConorWebb96 thanks for your answer. I am using a Database I have in MYSQL. So in the Form I have added a Option Picker that Selects the route we will be using to trasnport some cargo.. The idea is to keep a ledger of all the cargo moved, so the idea is that when we select the route we will be using, the form selects the correct distance stored for that route in the MYSQL DB. 

When I check my code with the 'TEXT' pf the rout name everyting works.. But as soon as I replace the Text for the Option Picker ID, the code stops working. 

{{#each Ruta.Rows }}
{{#if (eq Field Value this.Ruta)}}
{{this.Distancia}}
{{/if}}
{{/each}}

This is the current code I am using... If I replace Field Value by Let's. say 'BarranquillaSantamarta' everything works.

**Reply by ConorWebb96** [reply]
Hey @smtschampel,

Can you provide some screenshots of your component tree, plus fields/actions e.g. `update field value` that you are using. 

- Is your options picker set to custom? 
- Have you manually created the labels/values? For the above component.
- Is your option picker actually a relationship picker which you are then trying to use to pull in the distance stored within that route?
- When you say `field value`, do you actually mean `default value`? Instead of using that I'd recommend using the `update field action`. As the default value can only be set on page load/when the component is loaded in. 

Thanks in advance.

**Reply by smtschampel** [reply]
Hi @ConorWebb96 

I am using standard information stored on different tables to populate the form to reduce inputting wrong data. 

<img width="1860" alt="Screenshot 2023-09-25 at 12 16 07 PM" src="https://github.com/Budibase/budibase/assets/11296112/e55b7af7-0e5c-4ef3-b6cf-2fe0c61dbba3">


The information Origen/Destino is populated as we click the row Ruta on the TableBlock: SelectorRuta.

But some additional information must be populated via option selectors, for example, Flota, a selector connected to Flota_Data_Provider. 

That option picker should select the Flota to use, which has additional information on the same row, such as the Fuel Consumption, that should be available to populate different Form Fields.

For example, CombustibleVacio Field should take the fuel rate consumption stored for Flota XVV579 selected on the option picker. For this, I used the following code. 

<img width="1353" alt="Screenshot 2023-09-25 at 12 28 59 PM" src="https://github.com/Budibase/budibase/assets/11296112/f229f1e1-7051-4901-b964-450ce2915db8">

<img width="634" alt="Screenshot 2023-09-25 at 12 30 11 PM" src="https://github.com/Budibase/budibase/assets/11296112/ae69e7ad-c7d4-47b1-a4d1-80e951420183">

This is not working, but if I replace FliedValue for the string, it should look for. For example, 'XVV579' works perfectly, as shown in the following screenshot.

<img width="552" alt="Screenshot 2023-09-25 at 12 29 17 PM" src="https://github.com/Budibase/budibase/assets/11296112/05a5e35f-5f11-46bd-82c2-527370489f74">

How can I populate the form using the information linked to the row table of a selected option picker?

<img width="350" alt="Screenshot 2023-09-25 at 12 33 59 PM" src="https://github.com/Budibase/budibase/assets/11296112/ad865cdd-18f8-4406-857e-9ba4b6587fcc">
